apps/balancer-tools/src/app/apr/round/(components)/PoolListTable.tsx

```diff
@@ -30,22 +29,48 @@ interface PoolStats {
 interface PoolTableData extends PoolStats {
   id: string;
   network: number;
+  symbol: string;
 }
 
 export function PoolListTable({ roundId }: { roundId: string }) {
-  const initialTableValues: PoolTableData[] = votingGauges
-    .slice(0, 10)
-    .map((gauge) => ({
-      id: gauge.pool.id,
-      apr: 0,
-      balPriceUSD: 0,
-      tvl: 0,
-      votingShare: 0,
-      network: gauge.network,
-    }));
-  const [tableData, setTableData] = useState(initialTableValues);
+  const [isLoadingFetchPoolList, setIsLoadingFetchPoolList] = useState(true);
+  const [tableData, setTableData] = useState([] as PoolTableData[]);
   const [sortField, setSortField] = useState("");
   const [order, setOrder] = useState<"asc" | "desc">("asc");
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const poolList = await getPoolList();
+
+        //maybe round can be passed to getPoolList and calculatePoolStats can be done there
+        const poolListWithStats = await Promise.all(
+          poolList.map(async (pool) => {
+            const poolStats = await calculatePoolStats({
+              roundId,
+              poolId: pool.id,
+            });
+
+            return {
+              id: pool.id,
+              network: Number(pool.networkId),
+              symbol: pool.symbol as string,
+              ...poolStats,
+            };
+          }),
+        );
+
+        setTableData(poolListWithStats);
+
+        setIsLoadingFetchPoolList(false);
+      } catch (error) {
+        setIsLoadingFetchPoolList(false);
+      }
```

Comment:
❌ this must be done on the server side

Reply:
I need some help with this. Tried to use server actions but without success :(